src/lib/broadcast.jl

```diff
@@ -172,23 +172,25 @@ collapse_nothings(xs) = xs
 @adjoint function broadcasted(::AbstractArrayStyle, f, args...)
   len = inclen(args)
   y∂b = _broadcast((x...) -> _pullback(__context__, f, x...), args...)
-  y = map(x -> x[1], y∂b)
-  ∂b = map(x -> x[2], y∂b)
-  y, function (ȳ)
-    dxs_zip = map((∂b, ȳ) -> ∂b(ȳ), ∂b, ȳ)
-    dxs = collapse_nothings.(ntuple(i -> map(x -> _get(x, i), dxs_zip), len))
+  y = map(first, y∂b)
+  function ∇broadcasted(ȳ)
+    dxs_zip = map((pair, ȳ₁) -> last(pair)(ȳ₁), y∂b, ȳ)
+    dxs = ntuple(len) do i
+      collapse_nothings(map(StaticGetter{i}(), dxs_zip))
```

Comment:
We might need to add a overload to `StaticGetter{i}()(::Nothing) = nothing`?
Then we can maybe delete `_get` ?

We won't need that in ChainRules since `getindex` on `AbstractZero`s always returns a abstract zero.
Though we may want to add the trick from here for that 
https://github.com/JuliaDiff/ChainRules.jl/pull/441#discussion_r654373591
(Could also add that to Zygote for that matter)

Reply:
Oh, might be. I didn't run tests locally, perhaps CI will complain.

Reply:
CI did complain, method added now.

Reply:
don't forget to delete `_get` those functions are not used anywhere anymore